### PR TITLE
Return correct error on lock configuration check

### LINKED ISSUE #869: PutObjectLockConfiguration: invalid values should be handled properly
Test test_object_lock_put_obj_lock_invalid_status() tries to put invalid ObjectLock status:

```
"ObjectLockEnabled": "Disabled",
```

s3 gate responses with 500:

```
2023-10-19T23:30:02.919Z        error   handler/util.go:29      call method     {"status": 500, "request_id": "a65ed6d2-7742-4e1b-b8e7-ab9ce9334d7e", "method": "PutBucketObjectLockConfig", "bucket": "yournamehere-g5k624pbd3p00jfh-1", "object": "", "description": "invalid lock configuration", "error": "invalid ObjectLockEnabled value: Disabled"}
```

Response code should be 400. 

Same thing with other invalid config values, like mode (test_object_lock_put_obj_lock_invalid_mode) or years (test_object_lock_put_obj_lock_invalid_years) or days (test_object_lock_put_obj_lock_invalid_days)

### PR DESCRIPTION
**Pull request by smallhive**
Closes #869.

### CODE REVIEW

**Review by roman-khimov — DISMISSED** [review]

File: api/handler/locking.go [hunk 1]
```diff
@@ -287,15 +287,15 @@ func checkLockConfiguration(conf *data.ObjectLockConfiguration) error {
 
 	retention := conf.Rule.DefaultRetention
 	if retention.Mode != governanceMode && retention.Mode != complianceMode {
-		return fmt.Errorf("invalid Mode value: %s", retention.Mode)
+		return s3errors.GetAPIError(s3errors.ErrMalformedXML)
```
Comment (roman-khimov) [inline api/handler/locking.go:290]: Why not `GetAPIErrorWithError` to give a bit more details to the caller?

Comment (smallhive) [reply api/handler/locking.go]: Updated

File: api/handler/locking.go [hunk 2]
```diff
@@ -287,15 +287,15 @@ func checkLockConfiguration(conf *data.ObjectLockConfiguration) error {
...
 	retention := conf.Rule.DefaultRetention
 	if retention.Mode != governanceMode && retention.Mode != complianceMode {
-		return fmt.Errorf("invalid Mode value: %s", retention.Mode)
+		return s3errors.GetAPIErrorWithError(s3errors.ErrMalformedXML, fmt.Errorf("invalid Mode value: %s", retention.Mode))
 	}
 
-	if retention.Days == 0 && retention.Years == 0 {
-		return fmt.Errorf("you must specify Days or Years")
+	if retention.Days <= 0 && retention.Years <= 0 {
+		return s3errors.GetAPIError(s3errors.ErrMalformedXML)
```
Comment (roman-khimov) [inline api/handler/locking.go:294]: All the other ones can be improved too

Comment (smallhive) [reply api/handler/locking.go]: Improved

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**